### ISSUE TITLE
Accuracy suddenly Falling down

### ISSUE THREAD

**Issue by sudarshaana** [googlecolab/colabtools#413]
Hi, 
I was running a CNN on Google Colab. It was running as expected last time I run (on 29th January).

Yesterday I run the same code with 150 Epoch (As previous), but I was shocked seeing the accuracy has fallen down from 94 to 2% after around 14-22 Epoch and not improving.

I can understand what is happening. I run the code on my local pc (on CPU), it's working fine.

Can anyone have any suggestion?

**Comment by colaboratory-team** [comment]
Can you share a notebook that reproduces this problem? i.e., one which has good training accuracy with a TensorFlow install outside of Colab, but performs poorly in Colab itself.

**Comment by sudarshaana** [comment]
https://colab.research.google.com/drive/19zLZR1l-p1tlbd3sLUombqA9e7Vv1Nyo

**Comment by craigcitro** [comment]
I just ran a copy of this against a fresh runtime, and I'm not seeing the accuracy drop you're describing:

```
Epoch 1/100
75/75 [==============================] - 443s 6s/step - loss: 3.8220 - acc: 0.0393 - val_loss: 3.3939 - val_acc: 0.1337

Epoch 00001: val_loss improved from inf to 3.39392, saving model to weights.best.with_augmentation_new.hdf5
Epoch 2/100
75/75 [==============================] - 442s 6s/step - loss: 3.2295 - acc: 0.1456 - val_loss: 2.5557 - val_acc: 0.3451

Epoch 00002: val_loss improved from 3.39392 to 2.55575, saving model to weights.best.with_augmentation_new.hdf5
Epoch 3/100
75/75 [==============================] - 446s 6s/step - loss: 2.5581 - acc: 0.2915 - val_loss: 1.8299 - val_acc: 0.5114

Epoch 00003: val_loss improved from 2.55575 to 1.82985, saving model to weights.best.with_augmentation_new.hdf5
Epoch 4/100
75/75 [==============================] - 446s 6s/step - loss: 2.0176 - acc: 0.4233 - val_loss: 1.3202 - val_acc: 0.6078
```
Am I misunderstanding the problem you're hitting?

**Comment by craigcitro** [comment]
Ah, I think maybe I'm now seeing what you are:

```
Epoch 00013: val_loss improved from 0.37024 to 0.30073, saving model to weights.best.with_augmentation_new.hdf5
Epoch 14/100
75/75 [==============================] - 443s 6s/step - loss: 0.4158 - acc: 0.8766 - val_loss: 0.3119 - val_acc: 0.8966

Epoch 00014: val_loss did not improve from 0.30073
Epoch 15/100
75/75 [==============================] - 440s 6s/step - loss: nan - acc: 0.0658 - val_loss: nan - val_acc: 0.0229

Epoch 00015: val_loss did not improve from 0.30073

Epoch 00015: ReduceLROnPlateau reducing learning rate to 0.0005000000237487257.
Epoch 16/100
75/75 [==============================] - 439s 6s/step - loss: nan - acc: 0.0200 - val_loss: nan - val_acc: 0.0189

Epoch 00016: val_loss did not improve from 0.30073
Epoch 17/100
75/75 [==============================] - 441s 6s/step - loss: nan - acc: 0.0200 - val_loss: nan - val_acc: 0.0176
```

Is this the same drop in accuracy you were hitting?

**Comment by fchollet** [comment]
When training starts out fine, but eventually falls to a very low accuracy, it tends to indicate a training stability issue.

Upgrading your TF version changes the randomness behavior of your network, which can cause training stability issues to start manifesting themselves (they were already there, but you were lucky with the initialization).

Steps you can take to address training stability:
- Remove the `ReduceLROnPlateau` callback
- Reduce the initial learning rate
- Add gradient clipping to your optimizer
- Try to change the initializers of your layers
- Try a different optimizer (e.g. `RMSprop`)
- Make your network return logits and use `from_logits=True` in your loss function

**Comment by sudarshaana** [comment]
@craigcitro yes, I'm facing the same issue.

**Comment by sudarshaana** [comment]
I found the solution by downgrading the Tensorflow version. :)

**Comment by roaavisions** [comment]
I downgrading too but still not work my accuracy very very low please help me
